compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1101,24 +1071,95 @@ private STNode parseFuncDefOrFuncTypeDesc(STNode metadata, STNode visibilityQual
+        STNode funcDef = parseFuncDefOrMethodDeclEnd(metadata, visibilityQualifier, qualifiers, functionKeyword, name,
+                                                     resourcePath, funcSignature, isObjectMember, isObjectTypeDesc);
+        endContext();
+        return funcDef;
+    }
+
+    /**
+     * Parse function definition or function type descriptor identifier rhs.
+     *
+     * @param metadata Preceding metadata
+     * @param visibilityQualifier Preceding visibility qualifier
+     * @param qualifiers Preceding visibility qualifier
+     * @param functionKeyword `function` keyword
+     * @param isObjectMember Whether object member or not
+     * @param isObjectTypeDesc Whether object type or not
+     * @return Parsed node
+     */
+    private STNode parseFuncDefOrFuncTypeDescRhs(STNode name, STNode metadata, STNode visibilityQualifier,
+                                              List<STNode> qualifiers, STNode functionKeyword, boolean isObjectMember,
+                                              boolean isObjectTypeDesc) {
+        switch (peek().kind) {
+            case OPEN_PAREN_TOKEN:
+                STNode resourcePath = STNodeFactory.createEmptyNodeList();
+                return parseFunctionDefinition(metadata, visibilityQualifier, resourcePath, qualifiers, functionKeyword,
+                                               isObjectMember, isObjectTypeDesc, name);
+            case DOT_TOKEN:
+            case IDENTIFIER_TOKEN:
+            case OPEN_BRACKET_TOKEN:
+                resourcePath = parseRelativeResourcePath();
+                if (!isObjectMember) {
+                    addInvalidNodeToNextToken(resourcePath,
+                                              DiagnosticErrorCode.ERROR_RESOURCE_PATH_IN_FUNCTION_DEFINITION);
+                    resourcePath = STNodeFactory.createEmptyNodeList();
+                }
+                return parseFunctionDefinition(metadata, visibilityQualifier, resourcePath, qualifiers, functionKeyword,
+                                               isObjectMember, isObjectTypeDesc, name);
+            case EQUAL_TOKEN:
+            case SEMICOLON_TOKEN:
+                endContext();
+                functionKeyword = validateVisibilityQualifiersOfFunctionType(visibilityQualifier, qualifiers,
+                                                                             functionKeyword, isObjectMember);
+                List<STNode> extractQualifiersList = extractQualifiersOfFunctionType(qualifiers, isObjectMember,
+                                                                                     isObjectTypeDesc);
+                STNode qualifierNodeList = createFuncTypeQualNodeList(qualifiers, false);
+                STNode typeDesc = STNodeFactory.createFunctionTypeDescriptorNode(qualifierNodeList, functionKeyword,
+                                                                                 STNodeFactory.createEmptyNode());
```

Comment:
seems to be a duplicate of L1600-L1606.(code block inside the parseVarDeclWithFunctionType) 
Can extract to validateAndCreateFuncTyeDesc()?